### ISSUE TITLE
 property_exists(): must be of type object

### ISSUE THREAD

**Issue by Oxygenism** [Piggy-Loyalty/piggy-php-sdk#23]
Hi, I do not know if this issue is caused by legacy digital reward transactions or what. But it's an issue I've been running into with a few of the contacts in the store I'm working on a project for. Whenever I'm retrieving all the customers' loyalty transactions, I run into the following error:

Error:
`property_exists(): Argument #1 ($object_or_class) must be of type object|string, null given`

Trace:
```
#0 /var/www/html/vendor/piggy/piggy-php-sdk/src/Mappers/Loyalty/Rewards/DigitalRewardMapper.php(28): property_exists(NULL, 'active')
#1 /var/www/html/vendor/piggy/piggy-php-sdk/src/Mappers/Loyalty/Receptions/DigitalRewardReceptionMapper.php(34): Piggy\Api\Mappers\Loyalty\Rewards\DigitalRewardMapper->map(NULL)
#2 /var/www/html/vendor/piggy/piggy-php-sdk/src/Mappers/Loyalty/Receptions/RewardReceptionMapper.php(31): Piggy\Api\Mappers\Loyalty\Receptions\DigitalRewardReceptionMapper->map(Object(stdClass))
#3 /var/www/html/vendor/piggy/piggy-php-sdk/src/Mappers/Loyalty/LoyaltyTransactionMapper.php(45): Piggy\Api\Mappers\Loyalty\Receptions\RewardReceptionMapper->map(Object(stdClass))
#4 /var/www/html/vendor/piggy/piggy-php-sdk/src/Resources/OAuth/Loyalty/Receptions/LoyaltyTransactionsResource.php(43): Piggy\Api\Mappers\Loyalty\LoyaltyTransactionMapper->map(Array)
```

The 'digital_reward' and 'digital_reward_code' values on this transaction property are null.

I resolved the issue by applying the following patch:
```
Index: vendor/piggy/piggy-php-sdk/src/Mappers/Loyalty/Receptions/DigitalRewardReceptionMapper.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/vendor/piggy/piggy-php-sdk/src/Mappers/Loyalty/Receptions/DigitalRewardReceptionMapper.php b/vendor/piggy/piggy-php-sdk/src/Mappers/Loyalty/Receptions/DigitalRewardReceptionMapper.php
--- a/vendor/piggy/piggy-php-sdk/src/Mappers/Loyalty/Receptions/DigitalRewardReceptionMapper.php
+++ b/vendor/piggy/piggy-php-sdk/src/Mappers/Loyalty/Receptions/DigitalRewardReceptionMapper.php	(date 1714574740191)
@@ -31,8 +31,8 @@

         $contact = $contactMapper->map($data->contact);
         $shop = $shopMapper->map($data->shop);
-        $digitalReward = $digitalRewardMapper->map($data->digital_reward);
-        $digitalRewardCode = $digitalRewardCodeMapper->map($data->digital_reward_code);
+        $digitalReward = $data->digital_reward ? $digitalRewardMapper->map($data->digital_reward) : null;
+        $digitalRewardCode = $data->digital_reward ? $digitalRewardCodeMapper->map($data->digital_reward_code) : null;

         if (isset($data->contact_identifier)) {
             $contactIdentifier = $contactIdentifierMapper->map($data->contact_identifier);
Index: vendor/piggy/piggy-php-sdk/src/Models/Loyalty/Receptions/DigitalRewardReception.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/vendor/piggy/piggy-php-sdk/src/Models/Loyalty/Receptions/DigitalRewardReception.php b/vendor/piggy/piggy-php-sdk/src/Models/Loyalty/Receptions/DigitalRewardReception.php
--- a/vendor/piggy/piggy-php-sdk/src/Models/Loyalty/Receptions/DigitalRewardReception.php
+++ b/vendor/piggy/piggy-php-sdk/src/Models/Loyalty/Receptions/DigitalRewardReception.php	(date 1714574859715)
@@ -77,7 +77,7 @@
      * @param DigitalReward $digitalReward
      * @param DigitalRewardCode $digitalRewardCode
      */
-    public function __construct(string $type, int $credits, string $uuid, Contact $contact, Shop $shop, ?ContactIdentifier $contactIdentifier, DateTime $createdAt, string $title, DigitalReward $digitalReward, DigitalRewardCode $digitalRewardCode)
+    public function __construct(string $type, int $credits, string $uuid, Contact $contact, Shop $shop, ?ContactIdentifier $contactIdentifier, DateTime $createdAt, string $title, ?DigitalReward $digitalReward, ?DigitalRewardCode $digitalRewardCode)
     {
         $this->type = $type;
         $this->credits = $credits;
```

Not to say that, that is the optimal fix. But that's what resolved the error for me and my use case ;)

**Comment by iDiegoNL** [comment]
Thanks for opening an issue regarding this! I will be looking into this today.

You can expect a fix within a few days!

**Comment by iDiegoNL** [comment]
This issue has been fixed in #25 and has been released in [v3.4.0](https://github.com/Piggy-Loyalty/piggy-php-sdk/releases/tag/3.4.0). Thanks again for reporting this!
